### PR TITLE
ROX-18428: implement KMS encryption and decryption

### PR DESCRIPTION
**Pull request by johannes94**
## Description
Adding implementation and tests for En/Decryption of data using AWS KMS.

Change "RUN_AWS_RDS_TESTS"  to a more generic test suite "RUN_AWS_INTEGRATION". Also changed the github action workflow so that it now executes both KMS and RDS Integration tests.

## Checklist (Definition of Done)

- [x] Unit and integration tests added
- [x] Added test description under `Test manual`
- [x] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [x] CI and all relevant tests are passing
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [x] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- ~~[] Add secret to app-interface Vault or Secrets Manager if necessary~~
- ~~[ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)~~
- ~~[ ] Check AWS limits are reasonable for changes provisioning new resources~~

## Test manual

Integration tests are sufficient since this is not yet integrated into fleetshard-sync

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -327,12 +327,13 @@ test: $(GOTESTSUM_BIN)
...
-# Runs the AWS RDS integration tests.
-test/rds: $(GOTESTSUM_BIN)
-	RUN_RDS_TESTS=true \
-	$(GOTESTSUM_BIN) --junitfile data/results/rds-integration-tests.xml --format $(GOTESTSUM_FORMAT) -- -p 1 -v -timeout 45m -count=1 \
-		./fleetshard/pkg/central/cloudprovider/awsclient/...
-.PHONY: test/rds
+# Runs the AWS integration tests.
+test/aws: $(GOTESTSUM_BIN)
+	RUN_AWS_INTEGRATION=true \
+	$(GOTESTSUM_BIN) --junitfile data/results/aws-integration-tests.xml --format $(GOTESTSUM_FORMAT) -- -p 1 -v -timeout 45m -count=1 \
```
Comment (kurlov) [inline Makefile:333]: Why do we need timeout here if it's specified in the actions job?

**Review by kurlov — APPROVED** [review]
Looks great!

**Review by kurlov — APPROVED** [review]
Looks great!

File: Makefile (hunk 1 above)
Comment (johannes94) [reply Makefile:333]: In order to also timeout for local executions of this tests.

**Review by kurlov — COMMENTED** [review]
Looking good

**Pull request merged**